tap_google_search_console/sync.py

```diff
@@ -217,27 +214,23 @@ def sync_endpoint(client, #pylint: disable=too-many-branches
             data_dict = {}
             data_dict[data_key] = data_list
             data = data_dict
-        # LOGGER.info('data = {}'.format(data)) # TESTING, comment out
         if data_key in data:
-            # LOGGER.info('Number of raw data records: {}'.format(len(data[data_key])))
             transformed_data = transform_json(
                 data,
                 stream_name,
                 data_key,
                 site,
                 sub_type,
                 dimensions_list)[data_key]
-            # LOGGER.info('Number of transformed_data records: {}'.format(batch_count))
         else:
             LOGGER.info('Number of raw data records: 0')
-        # LOGGER.info('transformed_data = {}'.format(transformed_data))  # TESTING, comment out
         if not transformed_data or transformed_data is None:
             LOGGER.info('xxx NO TRANSFORMED DATA xxx')
             return 0 # No data results
         for record in transformed_data:
             for key in id_fields:
                 if not record.get(key):
-                    LOGGER.info('Missing key {} in record: {}'.format(key, record))
+                    raise ValueError('Missing key {} in record: {}'.format(key, record))
```

Comment:
Yes, absolutely remove the full record. If there's a non-PII value that can help with context, that can likely be logged, but full record is almost certainly not ok due to GDPR/PII, etc concerns.

Reply:
Is it alright if I print all the pk fields?

Reply:
Updated to only print the primary keys with values.

Reply:
And I tested it after reproducing the error by selecting false in menagerie for this field again.